src/VM/VMDetailsPrinter.swift

```diff
@@ -33,17 +33,25 @@ enum VMDetailsPrinter {
     
     /// Prints the status of all VMs in a formatted table
     /// - Parameter vms: Array of VM status objects to display
-    static func printStatus(_ vms: [VMDetails]) {
-        printHeader()
-        vms.forEach(printVM)
+    static func printStatus(_ vms: [VMDetails], json: Bool, print: (String) -> () = { print($0) }) throws {
```

Comment:
To test this I ended up using function-level dependency injection of the `print` global instance. Let me know if you'd do it differently.